### PR TITLE
Column Editor OK button should be disabled if Text-to-Insert is empty

### LINKED ISSUE #13315: Column Editor OK button should be disabled if Text-to-Insert is empty
### Description of the Issue
See StR.

### Steps to Reproduce (StR) the Issue
1. Invoke _Column Editor_
2. Click the _Text to Insert_ radio button
3. Delete all text out of the edit box below the _Text to Insert_ radio button

### Expected Behavior
I expected the OK button to turn be disabled/grey when the last character is removed from the edit box.

### Actual Behavior
OK button remains enabled:
image

### Debug Information
Notepad++ v8.5   (64-bit)
Build time : Mar  6 2023 - 02:18:13
Path : W:\npp.misc\_NoBackup\Releases\npp.8.5\npp.8.5.portable.x64\notepad++.exe
Command Line : -titleAdd="8.5 64-bit" -multiInst
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit)
OS Version : 21H2
OS Build : 19044.2486
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    PythonScript (2)

### PR DESCRIPTION
**Pull request by doug1234**
Fix #13315

### CODE REVIEW

File: PowerEditor/src/ScintillaComponent/columnEditor.cpp [hunk 1]
```diff
@@ -472,6 +473,23 @@ void ColumnEditorDlg::switchTo(bool toText)
 
 	::SetFocus(toText?hText:hNum);
 
+	if (toText) 
+	{
+		ColumnEditorParam colEditParam = NppParameters::getInstance()._columnEditParam;
+		if (!colEditParam._insertedTextContent.empty())
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/columnEditor.cpp:479]: It'll be more straight forward to get text from the text field control.

File: PowerEditor/src/ScintillaComponent/columnEditor.cpp [hunk 2]
```diff
@@ -472,6 +473,23 @@ void ColumnEditorDlg::switchTo(bool toText)
 
 	::SetFocus(toText?hText:hNum);
 
+	if (toText) 
+	{
+		ColumnEditorParam colEditParam = NppParameters::getInstance()._columnEditParam;
+		if (!colEditParam._insertedTextContent.empty())
+		{
+			::SetDlgItemText(_hSelf, IDC_COL_TEXT_EDIT, colEditParam._insertedTextContent.c_str());
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/columnEditor.cpp:481]: Why do we need this line?

File: PowerEditor/src/ScintillaComponent/columnEditor.cpp [hunk 3]
```diff
@@ -472,6 +473,23 @@ void ColumnEditorDlg::switchTo(bool toText)
...
+		}
+		else
+		{
+			::EnableWindow(::GetDlgItem(_hSelf, IDOK), false);
+		}
+	}
+	else 
+	{
+		::EnableWindow(::GetDlgItem(_hSelf, IDOK), true);
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/columnEditor.cpp:490]: EnableWindow(::GetDlgItem(_hSelf, IDOK), xxx); 
should be call only once at the end - xxx is the result of IDC_COL_TEXT_EDIT (empty or not).

**Review by donho — CHANGES_REQUESTED** [review]
Please make code simpler and straight forward.

**Comment by doug1234** [comment]
I simplified the change based on your comments.

File: PowerEditor/src/ScintillaComponent/columnEditor.cpp [hunk 4]
```diff
@@ -472,6 +473,18 @@ void ColumnEditorDlg::switchTo(bool toText)
 
 	::SetFocus(toText?hText:hNum);
 
+	if (toText) 
+	{
+		const int stringSize = 1024;
+		TCHAR str[stringSize];
```
Comment (ozone10) [inline PowerEditor/src/ScintillaComponent/columnEditor.cpp:479]: `const int stringSize = 1024;` can be replaced with `constexpr int stringSize = 1024;`
Add initializer to be sure e.g. `TCHAR str[stringSize]{};`

Comment (donho) [reply PowerEditor/src/ScintillaComponent/columnEditor.cpp:479]: @ozone10 
good remark.
Could you add the "best practice" in contribution coding style:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#coding-style 
Or create a issue for the "best practice" so I can add them in the guideline?

**Comment by donho** [comment]
Life is complicated already, let's make our code simpler.
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/81db72afb3bf579e2f63beaf2daee234be7005da

**Pull request closed without merging**